subsys/zigbee/cli/zigbee_cli_cmd_attr.c

```diff
@@ -352,21 +292,24 @@ int cmd_zb_readattr(const struct shell *shell, size_t argc, char **argv)
 		return -ENOEXEC;
 	}
 
-	req_data = &(entry->read_write_attr_req_data);
-	packet_info = &(req_data->packet_info);
+	req_data = &(entry->zcl_data.read_write_attr_req);
+	packet_info = &(entry->zcl_data.pkt_info);
 
 	packet_info->dst_addr_mode = parse_address(*(++argv), &(packet_info->dst_addr), ADDR_ANY);
 
 	if (packet_info->dst_addr_mode == ADDR_INVALID) {
 		zb_cli_print_error(shell, "Invalid address", ZB_FALSE);
-		return -EINVAL;
+		goto readattr_error;
```

Comment:
Just thinking - shall we move 290 above 285 and go to the ctx entry clearing routine in case of incorrect number of arguments?